plugins/resteasy-http/src/main/java/org/radargun/service/RESTEasyHTTPOperations.java

```diff
@@ -0,0 +1,226 @@
+package org.radargun.service;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.InetSocketAddress;
+import java.net.URLEncoder;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.traits.BasicOperations;
+
+/**
+ * Implementation of {@link BasicOperations} through the HTTP protocol, using the JAX-RS 2.0 client
+ * api with RESTEasy implementation.
+ *
+ * @author Alan Field &lt;afield@redhat.com&gt;
+ */
+public class RESTEasyHTTPOperations implements BasicOperations {
+   private static final Log log = LogFactory.getLog(RESTEasyHTTPOperations.class);
+   private final RESTEasyHTTPService service;
+
+   public RESTEasyHTTPOperations(RESTEasyHTTPService service) {
+      this.service = service;
+   }
+
+   @Override
+   public <K, V> HTTPCache<K, V> getCache(String cacheName) {
+      if (service.isRunning()) {
+         if (cacheName != null && (service.cacheName == null || !service.cacheName.equals(cacheName))) {
+            throw new UnsupportedOperationException();
+         }
+         return new HTTPCache<K, V>();
+      }
+      return null;
+   }
+
+   protected class HTTPCache<K, V> implements BasicOperations.Cache<K, V> {
+
+      @Override
+      public V get(K key) {
+         V result = null;
+         if (service.isRunning()) {
+            String target = buildUrl(service.cacheName, key);
+            try {
+               Invocation get = service.getHttpClient().target(target).request().accept(service.getContentType())
+                     .buildGet();
+               result = decodeByteArray(get.invoke(byte[].class));
+            } catch (NotFoundException e) {
+               log.warn("RESTEasyHTTPOperations.get::Key: " + key + " does not exist in cache: "
+                     + service.cacheName);
+            } catch (Exception e) {
+               throw new RuntimeException("RESTEasyHTTPOperations::get request threw exception: " + target, e);
+            }
+         }
+         return result;
+      }
+
+      @Override
+      public boolean containsKey(K key) {
+         if (service.isRunning()) {
```

Comment:
OPTIONS could be more suitable here, since we don't want to transport the whole value